docs/06/01/faq.md

```diff
@@ -0,0 +1,177 @@
+
+#### Do I need to implement all components in the adapter?
+
+No, you only need to implement the components you want to customize. The SDK will use its default components for any components not provided in your adapter.
+
+To use a combination of custom and default components, merge your implementations with the `defaultComponents` export ([View on GitHub](https://github.com/Gusto/embedded-react-sdk/blob/main/src/contexts/ComponentAdapter/adapters/defaultComponentAdapter.tsx)):
+
+```tsx
+import { defaultComponents } from '@gusto/embedded-react-sdk'
+
+const myAdapter = {
+  ...defaultComponents,
+  Button: props => <MyCustomButton {...props} />,
+  TextInput: props => <MyCustomTextInput {...props} />,
+}
+```
+
+### Design and Styling
+
+#### What's the difference between using the Component Adapter and just overriding the CSS?
+
+The Component Adapter gives you full control over the implementation of UI components, including their structure, behavior, and styling. Overriding CSS only allows you to modify the appearance of the default components.
+
+The Component Adapter is ideal when:
+
+- You want to maintain visual consistency with your existing React design system
+- You need to use specific UI component libraries that aren't compatible with the SDK's default styling
+- You need to modify the behavior of components beyond what's possible with styling
+
+#### How can I ensure my custom components maintain accessibility features?
+
+When implementing custom components, pay attention to:
+
+- Proper labeling of form controls
+- Appropriate ARIA attributes
+- Keyboard navigation
+- Focus management
+- Color contrast
+
+The prop interfaces include properties like `aria-describedby`, `isInvalid`, and others that support accessibility. Make sure your custom implementations use these props correctly.
+
+The SDK components follow accessibility best practices by default. When creating custom implementations, study the default components to understand how they handle accessibility concerns. The SDK uses [React Aria](https://react-spectrum.adobe.com/react-aria/) for accessible components, which you can reference for implementation details.
+
+### Implementing Components
+
+#### How do I implement components with complex behavior like ComboBox or DatePicker?
+
+For complex components, you have a few options:
+
+1. Use a third-party library that provides similar functionality
+2. Implement a simplified version that meets your specific needs
+3. Use the SDK's default implementation for complex components while customizing simpler ones
+
+For example, to implement a DatePicker with react-datepicker:
+
+```tsx
+import DatePicker from 'react-datepicker'
```

Comment:
I'm wondering in this case if it's sufficient to just point them to our default adapter for DatePicker given we already are demonstrating how to wire it up with react aria and if we link to the actual file it will be ensured to be up to date

Reply:
Let me know if you like this update